agent/agent.go

```diff
@@ -862,13 +880,26 @@ func (a *agent) runScript(ctx context.Context, lifecycle, script string) error {
 		}()
 	}
 
-	cmdPty, err := a.sshServer.CreateCommand(ctx, script, nil)
-	if err != nil {
-		return xerrors.Errorf("create command: %w", err)
-	}
-	cmd := cmdPty.AsExec()
 	cmd.Stdout = writer
 	cmd.Stderr = writer
+
+	start := time.Now()
+	defer func() {
+		end := time.Now()
+		execTime := end.Sub(start)
+		exitCode := 0
+		if err != nil {
+			exitCode = 255 // Unknown status.
+			var exitError *exec.ExitError
+			if xerrors.As(err, &exitError) {
+				exitCode = exitError.ExitCode()
+			}
+			logger.Warn(ctx, "script failed", slog.F("execution_time", execTime), slog.F("exit_code", exitCode), slog.Error(err))
+		} else {
+			logger.Info(ctx, "script completed", slog.F("execution_time", execTime), slog.F("exit_code", exitCode))
```

Comment:
nit: logger.Debug?

Reply:
I think it depends. How much do we want agent logs to output when not in verbose mode? This is an important part of e.g. the startup process, once that blocks other tasks in certain configurations, so I think info is good. But if we want more silence, then debug makes sense.

Reply:
Ok, maybe just add more context here as "script completed" sounds vague. _Startup script completed in time_?

Reply:
It's part of the logger via `lifecycle=startup` field.

Reply:
I admit that my intention is to make messages exposed to users more meaningful. `lifecycle=startup` is useful for filtering purposes, but if somebody is browsing logs just looking at messages, then `script completed` doesn't carry much information: _which script? is there only one script? is there any output stored in a file?_

Reply:
It's an admirable goal. I suppose we can do `fmt.Sprintf("%s script completed", lifecycle)`?

Reply:
SGTM!

BTW Please rebase against the main branch, so we're that the logs linter doesn't complain about this PR.